### Data Warehouse and data mining systems #2

In [1]:
import pandas as pd
import numpy as np 

DATA_DIR="datasets\\"

In [2]:
data1 = pd.read_csv("".join([DATA_DIR,"SpotifyTop200songs2017-2021\\data.csv"]), sep="#")
data2 = pd.read_excel("".join([DATA_DIR,"SpotifyTop200songs2017-2021\Spotify 2010 - 2019 Top 100 Songs.xlsx"]))
# print(data1.head(5))
# print('-'*120)
# print(data2.head(5))
print('-'*120)
print(data1.merge(data2, left_on='Track Name', right_on='title', how='outer').head(10).columns)
data_merged =data1.merge(data2, left_on='Track Name', right_on='title', how='outer')
print(data_merged.head(10))
print(data_merged.isnull().values.any())
merged = data_merged.loc[:,["Track Name"]].isnull().values.any(axis=1)
null_indexes = [i for i in range(len(merged)) if merged[i]]
# print(null_indexes)
# w pliku z hashami są brakujace nazwy pioseneek, do wywalenia
data_merged.loc[null_indexes,:].head(5)

------------------------------------------------------------------------------------------------------------------------
Index(['Position', 'Track Name', 'Artist', 'Streams', 'Date', 'Genre', 'title',
       'artist', 'top genre', 'year released', 'added', 'bpm', 'nrgy', 'dnce',
       'dB', 'live', 'val', 'dur', 'acous', 'spch', 'pop', 'top year',
       'artist type'],
      dtype='object')
   Position Track Name      Artist    Streams        Date  \
0       1.0    Starboy  The Weeknd  3135625.0  2017-01-01   
1       1.0    Starboy  The Weeknd  3342769.0  2017-01-02   
2       1.0    Starboy  The Weeknd  3563076.0  2017-01-03   
3       1.0    Starboy  The Weeknd  3619247.0  2017-01-04   
4       1.0    Starboy  The Weeknd  3579206.0  2017-01-05   
5       3.0    Starboy  The Weeknd  3590301.0  2017-01-06   
6       3.0    Starboy  The Weeknd  3460752.0  2017-01-07   
7       3.0    Starboy  The Weeknd  3127938.0  2017-01-08   
8       3.0    Starboy  The Weeknd  3364512.0  2017-01-

Position Track Name Artist    Streams        Date Genre title artist  \
90313       5.0        NaN    NaN  3568811.0  2017-07-20   NaN   NaN    NaN   
90314      13.0        NaN    NaN  2571960.0  2017-07-20   NaN   NaN    NaN   
90315      30.0        NaN    NaN  1798208.0  2017-07-20   NaN   NaN    NaN   
90316      39.0        NaN    NaN  1520291.0  2017-07-20   NaN   NaN    NaN   
90317       5.0        NaN    NaN  3653533.0  2017-07-21   NaN   NaN    NaN   

      top genre  year released  ... dnce  dB  live  val  dur  acous  spch  \
90313       NaN            NaN  ...  NaN NaN   NaN  NaN  NaN    NaN   NaN   
90314       NaN            NaN  ...  NaN NaN   NaN  NaN  NaN    NaN   NaN   
90315       NaN            NaN  ...  NaN NaN   NaN  NaN  NaN    NaN   NaN   
90316       NaN            NaN  ...  NaN NaN   NaN  NaN  NaN    NaN   NaN   
90317       NaN            NaN  ...  NaN NaN   NaN  NaN  NaN    NaN   NaN   

       pop  top year  artist type  
90313  NaN       NaN          NaN  
90314  NaN       NaN          NaN  
90315  NaN       NaN          NaN  
90316  NaN       NaN          NaN  
90317  NaN       NaN          NaN  

[5 rows x 23 columns]


#









<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Position</th>
      <th>Track Name</th>
      <th>Artist</th>
      <th>Streams</th>
      <th>Date</th>
      <th>Genre</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>1</td>
      <td>Starboy</td>
      <td>The Weeknd</td>
      <td>3135625</td>
      <td>2017-01-01</td>
      <td>['canadian pop', 'canadian contemporary r&amp;b', 'pop']</td>
    </tr>
    <tr>
      <th>1</th>
      <td>2</td>
      <td>Closer</td>
      <td>The Chainsmokers</td>
      <td>3015525</td>
      <td>2017-01-01</td>
      <td>['pop', 'pop dance', 'tropical house', 'edm', 'electropop', 'dance pop']</td>
    </tr>
    <tr>
      <th>2</th>
      <td>3</td>
      <td>Let Me Love You</td>
      <td>DJ Snake</td>
      <td>2545384</td>
      <td>2017-01-01</td>
      <td>['pop', 'electronic trap', 'dance pop', 'edm', 'pop dance', 'pop rap']</td>
    </tr>
    <tr>
      <th>3</th>
      <td>4</td>
      <td>Rockabye (feat. Sean Paul &amp; Anne-Marie)</td>
      <td>Clean Bandit</td>
      <td>2356604</td>
      <td>2017-01-01</td>
      <td>['pop', 'uk dance', 'dance pop', 'uk funky', 'tropical house', 'pop dance', 'post-teen pop', 'edm']</td>
    </tr>
    <tr>
      <th>4</th>
      <td>5</td>
      <td>One Dance</td>
      <td>Drake</td>
      <td>2259887</td>
      <td>2017-01-01</td>
      <td>['toronto rap', 'canadian pop', 'canadian hip hop', 'rap', 'pop rap', 'hip hop']</td>
    </tr>
  </tbody>
</table>